packages/studio-ui/src/components/AddPageButton/BasicPageDataCollector.tsx

```diff
@@ -53,19 +54,27 @@ export default function BasicPageDataCollector({
         setErrorMessage(validationResult.errorMessages.join("\r\n"));
         return false;
       }
-      try {
-        await handleConfirm(data.pageName, getPathValue);
-        return true;
-      } catch (err: unknown) {
-        if (err instanceof Error) {
-          setErrorMessage(err.message);
-          return false;
-        } else {
-          throw err;
-        }
+      if (isPagesJSRepo && !getPathValue) {
+        setErrorMessage("A getPath value is required.");
+        return false;
+      }
+      if (pages[data.pageName]) {
```

Comment:
Right now, we're doing validation of `BasicPageData` in two different ways. We have a `PageDataValidator` class that we use above and we have these individual checks here. We should consolidate all our logic in one place. Can we add these new checks into `PageDataValidator`?

Reply:
I also see that we check for an existing page of the same name in a validation step within `createPage`.  Should we keep this in just in case, or can we delete it?  Since we check here, we shouldn't have a duplicate page name by the time we hit `createPage`.

Reply:
I think we can delete the duplicate check in `addPage`.